// @applitools/execution-grid-client/dist/cli.js
#!/usr/bin/env node
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const yargs_1 = __importDefault(require("yargs"));
const proxy_server_1 = require("./proxy-server");
yargs_1.default
    .example([
    ['$ eg-client', 'Run Eyes Universal server on random port'],
    ['$ eg-client --port 8080', 'Run Eyes Universal server on port 8080'],
])
    .command({
    command: '*',
    builder: yargs => yargs.options({
        port: {
            description: 'run server on a specific port.',
            alias: 'p',
            type: 'number',
        },
        serverUrl: {
            description: 'run server with specific default eyes server url.',
            type: 'number',
        },
        apiKey: {
            description: 'run server with specific default api key.',
            type: 'number',
        },
    }),
    handler: async (args) => {
        const proxy = await (0, proxy_server_1.makeServer)(args);
        console.log(proxy.url);
    },
}).argv;
