// @applitools/execution-grid-client/dist/proxy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.proxy = void 0;
const http_proxy_1 = require("http-proxy");
const stream_1 = require("stream");
async function proxy(request, response, options) {
    const proxy = (0, http_proxy_1.createProxy)();
    return new Promise((resolve, reject) => {
        const content = options.body ? JSON.stringify(options.body) : undefined;
        const settings = {
            [options.forward ? 'forward' : 'target']: options.target,
            selfHandleResponse: !options.forward,
            ws: true,
            changeOrigin: true,
            buffer: content
                ? new stream_1.Readable({
                    read() {
                        this.push(content);
                        this.push(null);
                    },
                })
                : undefined,
            headers: content
                ? Object.assign(Object.assign({}, options.headers), { 'Content-Length': Buffer.byteLength(content).toString() }) : options.headers,
        };
        proxy.web(request, response, settings, reject);
        proxy.on('proxyRes', resolve);
    });
}
exports.proxy = proxy;
